src/rational/q/arithmetic/pow.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module provides an implementation of the [`Pow`] trait for [`Q`].
+
+use crate::{
+    error::MathError,
+    integer::Z,
+    macros::for_others::{implement_for_others, implement_for_owned},
+    rational::Q,
+    traits::Pow,
+};
+use flint_sys::fmpq::fmpq_pow_fmpz;
+
+impl Pow<&Z> for Q {
+    type Output = Q;
+
+    /// Raises the value of `self` to the power of an integer `exp`.
+    ///
+    /// Parameters:
+    /// - `exp`: specifies the exponent to which the value is raised
+    ///
+    /// Returns the value of `self` powered by `exp` as a new [`Q`] instance.
+    ///
+    /// # Example
+    /// ```
+    /// use qfall_math::{rational::Q, integer::Z};
+    /// use qfall_math::traits::*;
+    ///
+    /// let base = Q::try_from((&9, &1)).unwrap();
+    /// let exp = Z::from(3);
+    ///
+    /// let powered_value = base.pow(&exp).unwrap();
+    ///
+    /// assert_eq!(Q::try_from((&729, &1)).unwrap(), powered_value);
+    /// ```
```

Comment:
Can you add an example where a negative exponent is inserted that shows that the element has been inverted